### DISCUSSION TITLE
Adding new leagues in it

### DISCUSSION THREAD

**Discussion by PatWekan** [nickpadd/EuropeanFootballLeaguePredictor#71] [General]
Hello,

i put it in general due it is not a real issue or so it is more for me understanding.
What would need to be done to put on new leagues like dutch, turkish or so?
Or some cups etc.

Could it be done by us users or need to be set by the developer @nickpadd 

Thanks and great stuff to be honest :)

**Comment by nickpadd** [comment]
The model gathers data from multiple resources. In order to expand to more leagues the sources used to get all these data need to have the league you are interested in. The most important barrier is with getting advanced league table statistics using [Understat](https://understat.com/).

Understat provides league table statistics for the leagues available by the model ['EPL', 'SerieA', 'Ligue1', 'LaLiga', 'Bundesliga'] as well as Scottish Premier League which is not included in the model's predictions.

With minimal changes we could also include Sottish Premier League, but for other leagues this seems impossible while Understat does not provide statistics for them. Maybe in a major upcoming change the model could change data source for league table statistics and move to another website that includes more leagues, but as of now including other leagues seems impossible.

As for cups, there might be a way to include cup games but only for teams playing in the 1st tier of the leagues already available. In that way the model could interpret the cup game as another league game. 
But cup games seem more dependent on other factors like the upcoming league games, something that would require different modeling so it is definitely something difficult to achieve in the short term.

**Comment by PatWekan** [comment]
Hello

ok i see and fully understand: it was very interesting to get some backgrounds to understand what could be added and what is hard work to be done.

As i see in Understat, there is RFL which seems to be russian laeague:)

**Reply by nickpadd** [reply]
Yes. Mistake on my part, it is the Russian League.

Every idea could be helpful to further improve the model.
Thanks for the feedback.
